File: 2017 maj/c++/2.cpp
#include <iostream>
#include <fstream>
using namespace std;

int row[321];

bool pal()
{
    for (int i = 0; i < 160; i++)
        if (row[i] != row[319 - i])
            return false;
    return true;
}

int main()
{
    ifstream in("dane.txt");

    int ans = 0;
    for (int i = 0; i < 200; i++)
    {
        for (int j = 0; j < 320; j++)
        {
            in >> row[j];
        }
        ans += !pal();
    }

    cout << ans << '\n';
}

File: 2017 maj/python/2.py
with open("dane.txt", "r") as f:
    file = f.read().strip().split('\n')

def is_pal(l1, l2):
    for i, j in zip(l1, l2):
        if i != j:
            return False
    return True

ans = 0
for row in file:
    row = row.split()
    if not is_pal(row, row[::-1]):
        ans += 1
print(ans)